src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c

```diff
@@ -27,15 +30,208 @@ const ActorInit En_River_Sound_InitVars = {
     (ActorFunc)EnRiverSound_Update,
     (ActorFunc)EnRiverSound_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_River_Sound/EnRiverSound_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_River_Sound/EnRiverSound_Destroy.s")
+void EnRiverSound_Init(Actor* thisx, GlobalContext* globalCtx) {
+    /**
+     * Params format:
+     *     0xPPBB
+     *     P: Sence path index
+     *     B: Behavior
+    **/
+
+    THIS->b_14C = 0;
+    THIS->pathIndex = (thisx->params >> 8) & 0xFF;
+    thisx->params = thisx->params & 0xFF;
+
+    if (thisx->params >= 248) {
+        func_800F4870(thisx->params - 248);
+        Actor_Kill(thisx);
+    } else if (thisx->params == 247) {
+        func_800F6FB4(4);
+        Actor_Kill(thisx);
+    } else if (thisx->params == 12) {
+        u32 q = gSaveContext.inventory.questItems;
+        if (!(gBitFlags[12] & q) || (gBitFlags[14] & q))
+            Actor_Kill(thisx);
+    }
+}
+
+void EnRiverSound_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    if (thisx->params == 12)
+        func_800F50EC(&thisx->projectedPos);
+    else if (thisx->params == 13)
+        func_800F5504();
+}
+
+#if defined(NON_MATCHING)
+/*
+ * If anyone wants to try and figure this shit out, be my guest. I'm done.
+ */
```

Comment:
```suggestion
#ifdef NON_MATCHING
//If anyone wants to try and figure this shit out, be my guest. I'm done.
```
Is how our style does it.